bevy::picking::events::Drop clashes with std::ops::Drop

## What problem does this solve or what need does it fill?

Valid as of latest main (3cf70ba4f9d62396240b186ff379d27a312d2aa2).

#14686 Introduced a name clash when using `use bevy::prelude::*;`

- [`std::ops::Drop`](https://doc.rust-lang.org/stable/std/ops/trait.Drop.html) is an already well-known type in the rust std, and is in the std prelude.
- [`bevy::picking::events::Drop`](https://dev-docs.bevyengine.org/bevy/picking/events/struct.Drop.html) is included in bevy's prelude.

Which causes this error:

```
error[E0404]: expected trait, found struct `Drop`
 --> src\t.rs:5:6
  |
5 | impl Drop for Foo {
  |      ^^^^ not a trait
  |
help: consider importing one of these items instead
  |
1 + use core::ops::Drop;
  |
1 + use std::ops::Drop;
  |

```

with:

```rust
use bevy::prelude::*;

pub struct Foo {}

impl Drop for Foo {
    fn drop(&mut self) {}
}
```

## What solution would you like?

Rename `bevy::picking::events::Drop`. No personal preference on what the new name should be.

--- a/crates/bevy_picking/src/events.rs
+++ b/crates/bevy_picking/src/events.rs
@@ -26,7 +26,7 @@ pub struct Pointer<E: Debug + Clone + Reflect> {
     pub pointer_id: PointerId,
     /// The location of the pointer during this event
     pub pointer_location: Location,
-    /// Additional event-specific data. [`Drop`] for example, has an additional field to describe
+    /// Additional event-specific data. [`DragDrop`] for example, has an additional field to describe
     /// the `Entity` that is being dropped on the target.
     pub event: E,
 }
@@ -193,7 +193,7 @@ pub struct DragLeave {
 
 /// Fires when a pointer drops the `dropped` entity onto the `target` entity.
 #[derive(Clone, PartialEq, Debug, Reflect)]
-pub struct Drop {
+pub struct DragDrop {
     /// Pointer button lifted to drop.
     pub button: PointerButton,
     /// The entity that was dropped onto the `target` entity.
@@ -626,7 +626,7 @@ pub fn send_drag_over_events(
                 pointer_id,
                 pointer_location.clone(),
                 target,
-                Drop {
+                DragDrop {
                     button,
                     dropped: target,
                     hit: hit.clone(),

--- a/crates/bevy_picking/src/events.rs
+++ b/crates/bevy_picking/src/events.rs
@@ -517,7 +517,7 @@ pub fn send_drag_over_events(
     //  - Pointer<DragEnter>,
     //  - Pointer<DragOver>,
     //  - Pointer<DragLeave>,
-    //  - Pointer<Drop>,
+    //  - Pointer<DragDrop>,
     mut commands: Commands,
     // Input
     drag_map: Res<DragMap>,

